lib/ansible/module_utils/network/meraki/meraki.py

```diff
@@ -0,0 +1,408 @@
+        org_net = {'org_id': org_id,
+                    'net_id': net_id,
+                    }
+        built_path = self.url_catalog[action][self.function]
+
+        if org_id and net_id:
+            built_path = built_path.format(**org_net)
+        elif org_id and not net_id:
+            built_path = built_path.format(**org_only)
+        elif not org_id and net_id:
+            built_path = built_path.format(**net_only)
+        # self.fail_json(msg=built_path)
+        return built_path
+
+    # def construct_path(self, action, function=None, org_id=None, net_id=None, org_name=None, append=None):
+    #     org_only = {'org_id': org_id,
+    #                 }
+    #     net_only = {'net_id': net_id,
+    #                 }
+    #     org_net = {'org_id': org_id,
+    #                 'net_id': net_id,
+    #                 }
+
+    #     built_path = None
+    #     if function is None:
+    #         built_path = self.url_catalog[action][self.function]
+    #     else:
+    #         self.function = function
+    #         built_path = self.url_catalog[action][function]
+    #     if 'org_id' in built_path:  # TODO: This is a mess, fix it
+    #         if org_id is None:
+    #             built_path = built_path.format(org_id=self.get_org_id(org_name))
+    #         else:
+    #             built_path = built_path.format(org_id=str(org_id), net_id=None)
+    #     if 'net_id' in built_path:
+    #         if net_id is None:
+    #             built_path = built_path.format(net_id=self.get_net_id(self.module.params['net_name']))
+    #         else:
+    #             built_path = built_path.format(net_id=net_id)
+    #     return built_path
+
+    def request(self, path, method=None, payload=None):
+        ''' Generic HTTP method for Meraki requests '''
+        self.path = path
+        self.define_protocol()
+        # if self.define_method() is -1:  # No changes are needed to existing object
+        #     return
+
+        if method is not None:
+            self.method = method
+        self.url = '{0}://{1}/api/v0/{2}'.format(self.params['protocol'], self.params['host'], self.path.lstrip('/'))
+        resp, info = fetch_url(self.module, self.url,
+                               headers=self.headers,
+                               data=payload,
+                               method=self.method,
+                               timeout=self.params['timeout'],
+                               use_proxy=self.params['use_proxy'],
+                               )
+        self.response = info['msg']
+        self.status = info['status']
+
+        if self.status >= 400:
+            try:
+                self.error['text'] = self.response_json(info['body'])
+                self.error['code'] = info['status']
+                self.fail_json(msg='Dashboard API error %(code)s: %(text)s' % self.error)
+            except KeyError:
+                self.fail_json(msg='Connection failed for %(url)s. %(msg)s' % info)
```

Comment:
In fact you could have one big try-except statement, and only a single KeyError exception clause for the connection error.